sdks/python/apache_beam/io/filebasedsink.py

```diff
@@ -307,7 +314,7 @@ def _rename_batch(batch):
           return exceptions
 
       exception_batches = util.run_using_threadpool(
-          _rename_batch, zip(source_file_batch, destination_file_batch),
+          _rename_batch, list(zip(source_file_batch, destination_file_batch)),
```

Comment:
IIUC run_using_threadpool accepts any iterable, so forcing a list is redundant.

Reply:
Reading `util.run_using_threadpool`, it looks like the code calls `len(inputs)`, which may not be compatible with all iterables.

Reply:
Ack, please discard my initial comment.